frontend/components/CompactNavigationBar.js

```diff
@@ -177,260 +147,134 @@ const CompactNavigationBar = () => {
-              }
-            >
-              <a id="mediaText">Media ▸</a>
-            </div>
-          </div>
-          <div
-            className={
-              mediaDropdownOpen
-                ? styles.menuDropdownContent
-                : `${styles.menuDropdownContent} ${styles.hiddenMenuDropdownContent}`
+          {navigationBarContent.map((page, index) => {
+            const menuContentLength =
+              Object.keys(navigationBarContent).length - 1;
+            if (page.dropdownContent) {
+              const dropdownContentLength =
+                Object.keys(page.dropdownContent).length - 1;
+              return (
+                <div key={index}>
+                  <div
+                    className={styles.menuDropdownContainer}
+                    id={page.id}
+                    onClick={
+                      page.id === "aboutUsDropdown"
+                        ? aboutUsDropdownClick
+                        : mediaDropdownClick
+                    }
+                    key={index}
+                  >
+                    {/* apply dropdown underline and red text color accordingly:
+                        if the current page belongs to the dropdown, the text should be underlined
+                        if the dropdown is open, the text should be red */}
+                    <div
+                      className={
+                        page.id === "aboutUsDropdown"
+                          ? router.asPath.split("/")[1] === "about" &&
+                            aboutUsDropdownOpen
+                            ? `${styles.currentPageMenuUnderline} ${styles.currentDropdown} ${styles.menuItem}`
+                            : router.asPath.split("/")[1] === "about"
+                            ? `${styles.currentPageMenuUnderline} ${styles.menuItem}`
+                            : aboutUsDropdownOpen
+                            ? `${styles.currentDropdown} ${styles.menuItem}`
+                            : styles.menuItem
+                          : router.asPath.split("/")[1] === "media" &&
+                            mediaDropdownOpen
+                          ? `${styles.currentPageMenuUnderline} ${styles.currentDropdown} ${styles.menuItem}`
+                          : router.asPath.split("/")[1] === "media"
+                          ? `${styles.currentPageMenuUnderline} ${styles.menuItem}`
+                          : mediaDropdownOpen
+                          ? `${styles.currentDropdown} ${styles.menuItem}`
+                          : styles.menuItem
+                      }
+                      key={index}
+                    >
+                      <a id={page.text} key={index}>
+                        {page.pageName}
+                      </a>
+                    </div>
+                  </div>
+                  <div
+                    className={
+                      page.id === "aboutUsDropdown"
+                        ? aboutUsDropdownOpen
```

Comment:
Look into making lines 188-205 more readable/maintainable. Perhaps we can use if statements here instead of ternary.